### PR TITLE
Data format

### PR DESCRIPTION
**Pull request by jackcarlisle**
# ready for review

- [x] formats the data before merge

#379

### CODE REVIEW

File: test/lib/merger_test.exs [hunk 1]
```diff
@@ -119,4 +119,23 @@ defmodule Karma.MergerTest do
...
+    with_mocks([
+      {Karma.S3, [],
+       [download: fn(_, _) -> {:ok, "www.aws.someurl.pdf"} end,
+        upload: fn(_) -> {:ok, :url, "www.aws.another.pdf"} end]},
+      {Karma.ScriptRunner, [],
+       [run_merge_script: fn(_) -> {"destination.pdf", 0} end]}
+    ]) do
+      assert {:ok, "www.aws.another.pdf"} = Merger.merge(offer, document)
+    end
+  end
```
Comment (JMurphyWeb) [inline test/lib/merger_test.exs]: We should be able to unit test the formatting functions fairly easily without all the mocking. Hopefully when you split it out we'll be able to simplify this test and assert something meaningful.

File: lib/format.ex [hunk 2]
```diff
@@ -0,0 +1,130 @@
+defmodule Karma.Formatter do
+  alias Karma.ViewHelpers
+
+  # formats nested map of all data, prefixes and flattens
+  def format(data) do
```
Comment (JMurphyWeb) [inline lib/format.ex]: this

File: lib/format.ex [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      prefix = Atom.to_string(key)
+
+      Map.get(data, key) # offer, startpack, user or project
+      |> prefix_keys(prefix) # user_first_name
+      |> Map.merge(acc)
+    end)
+  end
+
+  # helper used by format function
+  defp prefix_keys(map, prefix) do
```
Comment (JMurphyWeb) [inline lib/format.ex]: and this

Comment (JMurphyWeb) [inline lib/format.ex]: can stay in the merge function as it is formatting that will only get done in that file

File: test/lib/merger_test.exs (hunk 1 above)
Comment (jackcarlisle) [inline test/lib/merger_test.exs]: Will unit test this in a subsequent PR

Comment (JMurphyWeb) [inline test/lib/merger_test.exs]: Yeah. In order to make the functionality a bit more re-usable I imagine we will want to split that function up into `def format_offer_fields(offer) do` etc.. or maybe nicer would be a `def format_currency_fields(map, fields) do` which would be less code and super easy to unit test.

Ideally that would go into this PR but understand the desire to get this merged now.

File: lib/format.ex [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+          %{
+            offer_data |
+            "box_rental_fee_per_week": "#{currency_symbol}#{format_number(offer_data.box_rental_fee_per_week)}",
+            "box_rental_cap": "#{currency_symbol}#{format_number(offer_data.box_rental_cap)}",
+          }
+        _not_true ->
+          offer_data
+      end
+    offer_data =
+      case offer_data.equipment_rental_required? do
```
Comment (JMurphyWeb) [inline lib/format.ex]: I think these `case`s could be done using `if`s which would be less code for same result.

**Pull request merged**